docs/dotnet/dotnet-http-metrics.md

```diff
@@ -0,0 +1,252 @@
+# Semantic Conventions for HTTP-relevant metrics emitted by .NET
+
+**Status**: [Experimental, Feature-freeze][DocumentStatus]
+
+This document defines semantic conventions for HTTP metrics emitted by .NET components and runtime.
+
+<!-- toc -->
+
+- [HTTP client](#http-client)
+  * [Metric: `http.client.request.duration`](#metric-httpclientrequestduration)
+  * [Metric: `http.client.open_connections`](#metric-httpclientopen_connections)
+  * [Metric: `http.client.connection.duration`](#metric-httpclientconnectionduration)
+  * [Metric: `http.client.request.time_in_queue`](#metric-httpclientrequesttime_in_queue)
+  * [Metric: `http.client.active_requests`](#metric-httpclientactive_requests)
+- [HTTP server](#http-server)
+  * [Metric: `http.server.request.duration`](#metric-httpserverrequestduration)
+  * [Metric: `http.server.active_requests`](#metric-httpserveractive_requests)
+
+<!-- tocstop -->
+
+## HTTP client
+
+All Http client metrics are reported by `System.Net.Http` meter.
+
+### Metric: `http.client.request.duration`
+
+Client request duration measures time it takes to receive response headers and does not include time to read response body.
+
+This metric follows common [http.client.request.duration](../http/http-metrics.md#metric-httpclientrequestduration) definition.
+
+Notes:
+
+- Meter name is `System.Net.Http`
+- Metric added in .NET 8.0
+- When `error.type` attribute is reported, it contains one of [HTTP Request errors](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs), a full exception type, or a string representation of received status code.
```

Comment:
```suggestion
- When the `error.type` attribute is reported, it contains one of [HTTP Request errors](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httprequesterror) in snake_case, a full exception type, or a string representation of received status code.
```

Reply:
@antonfirsov  can you remove `/en-us` from your suggestion.

Reply:
Will these docs be localized? If not, to me it makes more sense to land on an English API doc page from an English spec.

Reply:
> Will these docs be localized? If not, to me it makes more sense to land on an English API doc page from an English spec.

I can't speak for the team but GitHub only doc's are rarely localized. Good writing is far more important than localization because google translate is very good when the doc was written for MT (machine translation). I'm doing an edit pass for MT.

Reply:
I would assume that if a user is proficient enough in English to read a documentation as advanced as a *specification*, there is a high chance they would prefer to see English terminology everywhere and find localizations confusing.

OFC it's just an opinion from personal experience, not going to die on this hill.

Reply:
I changed the suggestion, because this is a discussion beyond the scope of the PR, but I remain skeptical in this field.

Reply:
> I changed the suggestion, because this goes way beyond the scope of the PR, but I remain skeptical in this field.

The rule to remove `/lang-region` is MSFT wide, not something you could change at the .NET level.